### PR TITLE
[E-ivan < T390] Implement from_nodes on map

### PR DESCRIPTION
**Pull request by imilinovic**
### Description

Implement map.from_nodes

### Pull request type

- [ ] Bugfix
- [X] Algorithm/Module
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

######################################

#### Module/Algorithm
- [ ] Core algorithm/module implementation
- [ ] [Query module](https://memgraph.com/docs/memgraph/reference-guide/query-modules) implementation
- [ ] Unit tests
- [ ] End-to-end tests
- [ ] Code documentation
- [ ] README short description
- [ ] Documentation on [memgraph/docs](https://github.com/memgraph/docs)
- [ ] Update GQLALchemy signatures in [query builder](https://github.com/memgraph/gqlalchemy/blob/main/gqlalchemy/graph_algorithms/query_builder.py  ) using [query module signature generator](https://github.com/memgraph/gqlalchemy/blob/main/scripts/query_module_signature_generator.py)

######################################

### CODE REVIEW

File: cpp/map_module/algorithm/map.cpp [hunk 1]
```diff
@@ -1 +1,45 @@
...
+#include "mgp.hpp"
+
+void Map::FromNodes(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::memory = memory;
+  const auto arguments = mgp::List(args);
+  const auto record_factory = mgp::RecordFactory(result);
+
+  try {
+    const auto &label{arguments[0].ValueString()};
+    const auto &property{arguments[1].ValueString()};
```
Comment (antoniofilipovic) [inline cpp/map_module/algorithm/map.cpp:12]: You don't need these two references here.

File: cpp/map_module/algorithm/map.cpp [hunk 2]
```diff
@@ -1 +1,45 @@
...
+    mgp::Map map_result{};
+
+    const auto &all_nodes = mgp::Graph(memgraph_graph).Nodes();
```
Comment (antoniofilipovic) [inline cpp/map_module/algorithm/map.cpp:15]: You don't need reference here too

File: cpp/map_module/algorithm/map.cpp [hunk 3]
```diff
@@ -1 +1,45 @@
...
+    for (const auto &node : all_nodes) {
```
Comment (antoniofilipovic) [inline cpp/map_module/algorithm/map.cpp:16]: You don't need reference here too.

File: e2e_correctness/map_test/test_from_nodes_1/input.cyp [hunk 4]
```diff
@@ -0,0 +1,3 @@
+CREATE (Oppenheimer:Movie {id:1, title:'Oppenheimer', released:2023, tagline:'Now I am become Death, the destroyer of worlds.'})
+CREATE (Barbie:Movie {id:2, title:"Barbie", released:2023, tagline:'I am a Barbie girl in the Barbie world.'})
+CREATE (Shawshank:Movie {id:3, title:'The Shawshank Redemption', released:1994, tagline:'Over the course of several years, two convicts form a friendship, seeking consolation and, eventually, redemption through basic compassion.'});
```
Comment (antoniofilipovic) [inline e2e_correctness/map_test/test_from_nodes_1/input.cyp:3]: These tests only test graph structure, unfortunately. They don't test output. We need to use for these ones regular e2e tests.

Comment (antoniofilipovic) [reply e2e_correctness/map_test/test_from_nodes_1/input.cyp:3]: That is if I am not wrong as it seems from the code you are only returning results not updating anything directly

**Review by antoniofilipovic — CHANGES_REQUESTED** [review]
I think you need e2e unfortunately to test

File: e2e_correctness/map_test/test_from_nodes_1/input.cyp (hunk 4 above)
Comment (imilinovic) [reply e2e_correctness/map_test/test_from_nodes_1/input.cyp:3]: I think its not possible to create tests for this procedure then, will explain on daily.

**Review by antoniofilipovic — APPROVED** [review]

**Pull request merged**